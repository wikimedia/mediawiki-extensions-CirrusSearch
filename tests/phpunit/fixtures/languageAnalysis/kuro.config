{
	"langCode":"ja",
	"withPlugins":[ "analysis-kuromoji" ],
	"extended_tests":true,
	"expected.lang":"kuro.core"
}
