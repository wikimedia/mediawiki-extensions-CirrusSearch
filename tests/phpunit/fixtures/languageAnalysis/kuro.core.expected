{
    "analyzer": {
        "keyword": {
            "filter": [
                "truncate_keyword"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "lowercase_keyword": {
            "filter": [
                "truncate_keyword",
                "lowercase"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "near_match": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "near_match_asciifolding": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "truncate_keyword",
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "plain": {
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "plain_search": {
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "prefix": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "prefix",
            "type": "custom"
        },
        "prefix_asciifolding": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "prefix",
            "type": "custom"
        },
        "short_text": {
            "filter": [
                "lowercase",
                "aggressive_splitting",
                "asciifolding_preserve"
            ],
            "tokenizer": "whitespace",
            "type": "custom"
        },
        "short_text_search": {
            "filter": [
                "lowercase",
                "aggressive_splitting"
            ],
            "tokenizer": "whitespace",
            "type": "custom"
        },
        "source_text_plain": {
            "char_filter": [
                "word_break_helper_source_text"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "source_text_plain_search": {
            "char_filter": [
                "word_break_helper_source_text"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "suggest": {
            "filter": [
                "lowercase",
                "suggest_shingle"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "suggest_reverse": {
            "filter": [
                "lowercase",
                "suggest_shingle",
                "reverse"
            ],
            "tokenizer": "standard",
            "type": "custom"
        },
        "text": {
            "char_filter": [
                "globo_norm",
                "regex_acronym_fixer",
                "regex_camelCase",
                "word_break_helper",
                "arabic_extended_norm",
                "fullwidthnumfix",
                "kuromoji_combo_filter"
            ],
            "filter": [
                "decimal_digit",
                "kuromoji_baseform",
                "kuromoji_posfilter",
                "cjk_width",
                "ja_stop",
                "kuromoji_stemmer",
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "kuromoji_tokenizer",
            "type": "custom"
        },
        "text_search": {
            "char_filter": [
                "globo_norm",
                "regex_acronym_fixer",
                "regex_camelCase",
                "word_break_helper",
                "arabic_extended_norm",
                "fullwidthnumfix",
                "kuromoji_combo_filter"
            ],
            "filter": [
                "decimal_digit",
                "kuromoji_baseform",
                "kuromoji_posfilter",
                "cjk_width",
                "ja_stop",
                "kuromoji_stemmer",
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "kuromoji_tokenizer",
            "type": "custom"
        },
        "token_reverse": {
            "filter": [
                "reverse"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "trigram": {
            "filter": [
                "lowercase"
            ],
            "tokenizer": "trigram",
            "type": "custom"
        },
        "word_prefix": {
            "filter": [
                "lowercase",
                "prefix_ngram_filter"
            ],
            "tokenizer": "standard",
            "type": "custom"
        }
    },
    "char_filter": {
        "arabic_extended_norm": {
            "type": "mapping",
            "mappings": [
                "\\uFB8E=>\\u0643",
                "\\uFB8F=>\\u0643",
                "\\uFB90=>\\u0643",
                "\\uFB91=>\\u0643",
                "\\u06A9=>\\u0643",
                "\\u06AA=>\\u0643",
                "\\uFEDB=>\\u0643",
                "\\uFEDC=>\\u0643",
                "\\uFED9=>\\u0643",
                "\\uFEDA=>\\u0643",
                "\\uFBFC=>\\u064A",
                "\\uFBFD=>\\u064A",
                "\\uFBFE=>\\u064A",
                "\\uFBFF=>\\u064A",
                "\\u06CC=>\\u064A",
                "\\uFBE8=>\\u064A",
                "\\uFBE9=>\\u064A",
                "\\uFEEF=>\\u064A",
                "\\uFEF0=>\\u064A",
                "\\u0649=>\\u064A",
                "\\u06CD=>\\u064A",
                "\\uFBE4=>\\u064A",
                "\\uFBE5=>\\u064A",
                "\\uFBE6=>\\u064A",
                "\\uFBE7=>\\u064A",
                "\\u06D0=>\\u064A",
                "\\uFBA6=>\\u0647",
                "\\uFBA7=>\\u0647",
                "\\uFBA8=>\\u0647",
                "\\uFBA9=>\\u0647",
                "\\u06C1=>\\u0647",
                "\\u06C0=>\\u0647",
                "\\uFBA4=>\\u0647",
                "\\uFBA5=>\\u0647",
                "\\u06D5=>\\u0647"
            ]
        },
        "dotted_I_fix": {
            "type": "mapping",
            "mappings": [
                "\u0130=>I"
            ]
        },
        "fullwidthnumfix": {
            "type": "mapping",
            "mappings": [
                "\\uff10=>0",
                "\\uff11=>1",
                "\\uff12=>2",
                "\\uff13=>3",
                "\\uff14=>4",
                "\\uff15=>5",
                "\\uff16=>6",
                "\\uff17=>7",
                "\\uff18=>8",
                "\\uff19=>9"
            ]
        },
        "globo_norm": {
            "type": "mapping",
            "mappings": [
                "`=>'",
                "\u00b4=>'",
                "\u02b9=>'",
                "\u02bb=>'",
                "\u02bc=>'",
                "\u02bd=>'",
                "\u02be=>'",
                "\u02bf=>'",
                "\u02cb=>'",
                "\u055a=>'",
                "\\u05F3=>'",
                "\u2018=>'",
                "\u2019=>'",
                "\u201b=>'",
                "\u2032=>'",
                "\u2035=>'",
                "\ua78c=>'",
                "\uff07=>'",
                "\uff40=>'",
                "\\u202F=>\\u0020",
                "\u02c8=>",
                "\u02cc=>",
                "\\u0640=>",
                "\u066c=>,",
                "\u060c=>,",
                "\u055b=>",
                "\u055c=>",
                "\u055e=>",
                "\u00b5=>\u03bc",
                "\\u05F0=>\\u05D5\\u05D5",
                "\\u05F1=>\\u05D5\\u05D9",
                "\\u05F2=>\\u05D9\\u05D9",
                "\\uFB1F=>\\u05D9\\u05D9\\u05B7",
                "\\u05D9\\u05B7\\u05D9=>\\u05D9\\u05D9\\u05B7",
                "\u0d23\u0d4d\\u200D=>\u0d7a",
                "\u0d28\u0d4d\\u200D=>\u0d7b",
                "\u0d30\u0d4d\\u200D=>\u0d7c",
                "\u0d32\u0d4d\\u200D=>\u0d7d",
                "\u0d33\u0d4d\\u200D=>\u0d7e",
                "\u09a4\u09cd\\u200D=>\u09ce"
            ]
        },
        "kuromoji_combo_filter": {
            "type": "pattern_replace",
            "pattern": "[\\u0300-\\u0362]",
            "replacement": ""
        },
        "near_space_flattener": {
            "type": "mapping",
            "mappings": [
                "'=>\\u0020",
                "\\u2019=>\\u0020",
                "\\u02BC=>\\u0020",
                "_=>\\u0020",
                "-=>\\u0020"
            ]
        },
        "nnbsp_norm": {
            "type": "mapping",
            "mappings": [
                "\\u202F=>\\u0020"
            ]
        },
        "regex_acronym_fixer": {
            "type": "pattern_replace",
            "pattern": "(?<=(?:^|\\P{L})\\p{L})[.\uff0e](\\p{L})(?=\\P{L}|$)",
            "replacement": "$1"
        },
        "regex_camelCase": {
            "type": "pattern_replace",
            "pattern": "(\\p{Ll}[\\p{M}\\p{Cf}]*)([\\p{Lu}\\p{Lt}])",
            "replacement": "$1 $2"
        },
        "word_break_helper": {
            "type": "mapping",
            "mappings": [
                "_=>\\u0020",
                ":=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020",
                "\uff0e=>\\u0020",
                "\uff3f=>\\u0020",
                "\uff1a=>\\u0020",
                "\u00b7=>\\u0020"
            ]
        },
        "word_break_helper_source_text": {
            "type": "mapping",
            "mappings": [
                "_=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020",
                ":=>\\u0020"
            ]
        }
    },
    "filter": {
        "aggressive_splitting": {
            "type": "word_delimiter_graph",
            "stem_english_possessive": false,
            "preserve_original": false
        },
        "asciifolding": {
            "type": "asciifolding",
            "preserve_original": false
        },
        "asciifolding_preserve": {
            "type": "asciifolding",
            "preserve_original": true
        },
        "kuromoji_posfilter": {
            "type": "kuromoji_part_of_speech",
            "stoptags": [
                "\u63a5\u7d9a\u8a5e",
                "\u52a9\u8a5e",
                "\u52a9\u8a5e-\u683c\u52a9\u8a5e",
                "\u52a9\u8a5e-\u683c\u52a9\u8a5e-\u4e00\u822c",
                "\u52a9\u8a5e-\u683c\u52a9\u8a5e-\u5f15\u7528",
                "\u52a9\u8a5e-\u683c\u52a9\u8a5e-\u9023\u8a9e",
                "\u52a9\u8a5e-\u63a5\u7d9a\u52a9\u8a5e",
                "\u52a9\u8a5e-\u4fc2\u52a9\u8a5e",
                "\u52a9\u8a5e-\u526f\u52a9\u8a5e",
                "\u52a9\u8a5e-\u9593\u6295\u52a9\u8a5e",
                "\u52a9\u8a5e-\u4e26\u7acb\u52a9\u8a5e",
                "\u52a9\u8a5e-\u7d42\u52a9\u8a5e",
                "\u52a9\u8a5e-\u526f\u52a9\u8a5e\uff0f\u4e26\u7acb\u52a9\u8a5e\uff0f\u7d42\u52a9\u8a5e",
                "\u52a9\u8a5e-\u9023\u4f53\u5316",
                "\u52a9\u8a5e-\u526f\u8a5e\u5316",
                "\u52a9\u8a5e-\u7279\u6b8a",
                "\u52a9\u52d5\u8a5e",
                "\u8a18\u53f7",
                "\u8a18\u53f7-\u8aad\u70b9",
                "\u8a18\u53f7-\u53e5\u70b9",
                "\u8a18\u53f7-\u7a7a\u767d",
                "\u8a18\u53f7-\u62ec\u5f27\u958b",
                "\u8a18\u53f7-\u62ec\u5f27\u9589",
                "\u975e\u8a00\u8a9e\u97f3"
            ]
        },
        "lowercase": {
            "type": "lowercase"
        },
        "prefix_ngram_filter": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "remove_empty": {
            "type": "length",
            "min": 1
        },
        "suggest_shingle": {
            "type": "shingle",
            "min_shingle_size": 2,
            "max_shingle_size": 3,
            "output_unigrams": true
        },
        "truncate_keyword": {
            "type": "truncate",
            "length": 5000
        }
    },
    "tokenizer": {
        "no_splitting": {
            "type": "keyword"
        },
        "prefix": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "trigram": {
            "type": "nGram",
            "min_gram": 3,
            "max_gram": 3
        }
    }
}